### PR TITLE
Stops you from cutting a piece off of cable coils from a distance

### PR DESCRIPTION
**Pull request by goonstation-enjoyer**
[GAME OBJECTS] [BUG]
## About the PR 
Title. I tested this.

## Why's this needed? 
You shouldn't be able to cut a piece off of cables that are far away from you.

I don't think this needs a changelog.

### CODE REVIEW

**Comment by TobleroneSwordfish** [comment]
How is `attackby` getting called at a distance anyway?

File: code/obj/item/cable_coil.dm [hunk 1]
```diff
@@ -236,6 +236,10 @@ obj/item/cable_coil/dropped(mob/user)
 	if (issnippingtool(W) && src.amount > 1)
 		var/obj/item/cable_coil/A = split_stack(round(input("How long of a wire do you wish to cut?","Length of [src.amount]",1) as num))
 		if (istype(A))
+			if (GET_DIST(user, src) > 1)
```
Comment (ZeWaka) [inline code/obj/item/cable_coil.dm:239]: no. in interact range

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Comment by goonstation-enjoyer** [comment]
> How is `attackby` getting called at a distance anyway?

It isn't
1. Click cable with wirecutters
2. Drop cable and walk away
3. Put a number into the input box it opens
4. The cable teleports to you

File: code/obj/item/cable_coil.dm [hunk 2]
```diff
@@ -236,6 +236,10 @@ obj/item/cable_coil/dropped(mob/user)
 	if (issnippingtool(W) && src.amount > 1)
 		var/obj/item/cable_coil/A = split_stack(round(input("How long of a wire do you wish to cut?","Length of [src.amount]",1) as num))
 		if (istype(A))
+			if (!in_interact_range(src, user))
+				boutput(user, "You're too far away from the cable that you're trying to cut from!")
+				src.stack_item(A) //refund
+				return
 			A.set_loc(user.loc) //Hey, split_stack, Why is the default location for the new item src.loc which is *very likely* to be a damn mob?
```
Comment (ZeWaka) [inline code/obj/item/cable_coil.dm:243]: would be cleaner to just split stack after the check

Comment (ZeWaka) [reply code/obj/item/cable_coil.dm]: No, I meant modify the input() line so you don't split it until you actually do

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**